### PR TITLE
Restore homogeneous dataset collection indicator

### LINKED ISSUE #13744: Restore displaying homogenous dataset collection indicator
We added this in https://github.com/galaxyproject/galaxy/pull/13068, but that got dropped during the refactoring.  We need to add this feature back.

### PR DESCRIPTION
**Pull request by davelopez**
Fixes #13744

<del>I had to move the `elements_datatype` serialization field from the `detailed` to the `summary` view so it can be rendered in the history items without making additional requests. Hope that's ok.</del> See https://github.com/galaxyproject/galaxy/pull/13772#issuecomment-1106246781

Also, I slightly changed the description string (compared to the old history) depending on the `collection_type`, for example:
`list:paired` (heterogeneous): *a list of pairs with 10 items*  -> *a list with 10 pairs*
`list:paired` (homogeneous): *a list of pairs with 10 items*  -> *a list with 10 **tabular** pairs*

There is a complete [list of samples in the tests](https://github.com/galaxyproject/galaxy/compare/dev...davelopez:restore_homogeneous_hdca_indicator?expand=1#diff-a68e5baf968484751bfe46e55652d7ae3fa46c57a4abd21b08d5a8f3abcec32eR19-R58). I'm happy to further tweak this if neccesary.

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] Instructions for manual testing are as follows:
  - Create some collections with the same (and different) types of datasets
  - Observe the collection description matches the contents

Screenshot from 2022-04-20 14-55-55

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/components/History/Content/Collection/CollectionDescription.test.js [hunk 1]
```diff
@@ -11,23 +11,50 @@ describe("CollectionDescription", () => {
...
         await wrapper.setProps({ collectionType: "list:paired" });
-        expect(details.at(0).text()).toBe("a list of pairs");
+        expect(wrapper.text()).toBe(`a list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} pairs`);
+
+        await wrapper.setProps({ collectionType: "list:list" });
+        expect(wrapper.text()).toBe(`a list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} lists`);
+
         await wrapper.setProps({ collectionType: "other" });
-        expect(details.at(0).text()).toBe("a nested list");
+        expect(wrapper.text()).toBe(`a nested list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} datasets`);
```
Comment (mvdbeek) [inline client/src/components/History/Content/Collection/CollectionDescription.test.js]: These aren't datasets, `other` would be a deeply nested collection.
```suggestion
        expect(wrapper.text()).toBe(`a nested list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} dataset collections`);
```

Comment (davelopez) [reply client/src/components/History/Content/Collection/CollectionDescription.test.js]: Thanks for the catch Marius, I've rebased with the changes.

**Comment by bgruening** [comment]
Failing test seems to be related.

**Comment by dannon** [comment]
@davelopez Out of curiosity, do you have any metrics for how much more expensive it makes that summary request?

**Comment by davelopez** [comment]
Nope, I haven't taken any metrics yet. But if I recall well it was in the `betawebclient` view before and that's why it was working in the *old*beta history so I assume it's ok to have it in the summary unless we want to create another custom view to include it. That would probably take care of those failing framework tests which, by the way, I'm having trouble figuring out what's the real source of the error (apart from them not liking the `elements_datatypes`)

**Comment by davelopez** [comment]
In the backend chat, we decided to move the `elements_datatypes` to the dataset collection serialization when requesting `application/vnd.galaxy.history.contents.stats+json` media type (the one used by the History) so it won't interfere with existing API consumers for the `summary` view.

**Comment by davelopez** [comment]
All green! :tada:

**Review by mvdbeek — APPROVED** [review]
Very nice!

**Comment by dannon** [comment]
Nice, and I like the solution to the summary dict addition -- clever!

**Pull request merged**